core/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -148,7 +148,8 @@ public void onFailure(String source, Exception e) {
 
             @Override
             public boolean mustAck(DiscoveryNode discoveryNode) {
-                return discoveryNode.isMasterNode();
+                // repository is created on both master and data nodes
+                return discoveryNode.isMasterNode() || discoveryNode.isDataNode();
```

Comment:
Great catch! We must have missed that when we merged RepositoryService and ShardRepositoryService.